cpp/src/interop/to_arrow_device.cu

```diff
@@ -266,21 +267,28 @@ int dispatch_to_arrow_device::operator()<cudf::dictionary32>(cudf::column&& colu
                                                              ArrowArray* out)
 {
   nanoarrow::UniqueArray tmp;
+
+  cudf::dictionary_column_view const dview{column.view()};
+
   NANOARROW_RETURN_NOT_OK(initialize_array(
     tmp.get(),
-    id_to_arrow_type(column.child(cudf::dictionary_column_view::indices_column_index).type().id()),
+    dview.is_empty() ? NANOARROW_TYPE_INT32 : id_to_arrow_type(dview.indices().type().id()),
     column));
   NANOARROW_RETURN_NOT_OK(ArrowArrayAllocateDictionary(tmp.get()));
 
   auto contents = column.release();
   NANOARROW_RETURN_NOT_OK(set_null_mask(contents, tmp.get()));
 
   auto indices_contents =
-    contents.children[cudf::dictionary_column_view::indices_column_index]->release();
+    dview.is_empty()
```

Comment:
Just double-checking to be safe:  We're accessing `dview` here, despite having called `column.release()` just before it.  

That's safe to do, because the result of `column.release()` is kept alive in `contents`?

Reply:
(The alternative would have been to save away the result of `dview.is_empty()`.)

Reply:
Ah, good catch.